src/sys/unix.rs

```diff
@@ -1773,6 +1803,41 @@ impl crate::Socket {
                 })
         }
     }
+
+    /// Attach Berkeley Packet Filter(BPF) on this socket.
+    ///
+    /// BPF allows a user-space program to attach a filter onto any socket
+    /// and allow or disallow certain types of data to come through the socket.
+    ///
+    /// For more information about this option, see [filter](https://www.kernel.org/doc/html/v5.12/networking/filter.html)
+    #[cfg(all(feature = "all", target_os = "linux"))]
+    pub fn attach_filter(&self, mut filters: Vec<SockFilter>) -> io::Result<()> {
```

Comment:
We don't need ownership of the `filter`s, use `&[libc::sock_filter]` instead and we can create the `libc::sock_fprog` structure based on that.